tests/foreman/api/test_capsule.py

```diff
@@ -74,6 +74,15 @@ def test_positive_update_capsule(target_sat, module_capsule_configured):
     assert capsule.url in [cps.url for cps in capsules]
     assert capsule.name in [cps.name for cps in capsules]
 
+    @request.addfinalizer
+    def _finalize():
+        if pytestconfig.option.n_minus:
+            # Reinitializing the hostname
+            if cap := target_sat.api.SmartProxy().search(query={'search': f'name = {new_name}'}):
+                cap = cap[0]
+                cap.name = module_capsule_configured.hostname
+                cap.update(['name'])
+
```

Comment:
Can we move this finalizer method to top, if in case this test fails this teardown could get skip as its in end, right?